File: android/app/src/main/java/com/hindtechgroup/hindpharma/View/SuperAdminDashboardView.kt
package com.hindtechgroup.hindpharma.View

import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import androidx.lifecycle.viewmodel.compose.viewModel
import com.hindtechgroup.hindpharma.Models.SessionUser
import com.hindtechgroup.hindpharma.ViewModel.SuperAdminDashboardViewModel

@Composable
fun SuperAdminDashboardView(user: SessionUser, onBack: () -> Unit, viewModel: SuperAdminDashboardViewModel = viewModel()) {
    LaunchedEffect(user.token) { viewModel.load(user.token) }
    Column(Modifier.fillMaxSize().padding(16.dp)) {
        Row(Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween) {
            Text("HTG Super Admin", style = MaterialTheme.typography.headlineSmall)
            TextButton(onClick = onBack) { Text("Home") }
        }
        viewModel.errorMessage?.let { Text(it, color = MaterialTheme.colorScheme.error) }
        if (viewModel.isLoading) LinearProgressIndicator(Modifier.fillMaxWidth())
        Text("Businesses: ${viewModel.businesses.size}", style = MaterialTheme.typography.titleMedium, modifier = Modifier.padding(vertical = 12.dp))
        LazyColumn(verticalArrangement = Arrangement.spacedBy(8.dp)) {
            items(viewModel.businesses, key = { it.id }) { business ->
                Card(Modifier.fillMaxWidth()) {
                    Column(Modifier.padding(14.dp)) {
                        Text(business.businessName, style = MaterialTheme.typography.titleMedium)
                        Text("Admin: ${business.username}")
                        Text(if (business.isActive) "ACTIVE" else "INACTIVE")
                        business.subscriptionExpiry?.let { Text("Expiry: $it", style = MaterialTheme.typography.bodySmall) }
                    }
                }
            }
        }
    }
}

File: ios/HindPharmaApp/View/SuperAdminDashboardView.swift
import SwiftUI

struct SuperAdminDashboardView: View {
    let user: SessionUser
    @StateObject private var vm = SuperAdminDashboardViewModel()
    var body: some View {
        List {
            if let error = vm.errorMessage { Text(error).foregroundStyle(.red) }
            Section("HTG Businesses") {
                Text("Businesses: \(vm.admins.count)").font(.headline)
                ForEach(vm.admins) { admin in
                    VStack(alignment: .leading, spacing: 4) {
                        Text(admin.business_name).font(.headline)
                        Text("Admin: \(admin.username)")
                        Text(admin.is_active == 1 ? "ACTIVE" : "INACTIVE")
                        if let expiry = admin.subscription_expiry {
                            Text("Expiry: \(expiry)").font(.footnote).foregroundStyle(.secondary)
                        }
                    }
                    .padding(.vertical, 4)
                }
            }
        }
        .overlay { if vm.isLoading { ProgressView() } }
        .navigationTitle("HTG Super Admin")
        .task { await vm.load(token: user.token) }
    }
}
